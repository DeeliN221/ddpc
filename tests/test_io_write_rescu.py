"""Tests for the write functions in ddpc.io.read.dspaw_as."""

from pathlib import Path

from ddpc.io.read import rescu_xyz as rrx
from ddpc.io.write import rescu_xyz as wrx


def test_write(snapshot):
    """Tests reading various as files."""
    for s in Path(f"{__file__}/../structures").resolve().iterdir():
        if s.name.endswith(".xyz"):
            original_lines = s.read_text().split("\n")
            atoms = rrx.read(s)
            write_lines = wrx.write("-", atoms)
            for line, write_line in zip(original_lines, write_lines.split("\n"), strict=True):
                if line.startswith("AtomType"):  # ignore comment line
                    continue
                # compare them, ignore white spaces
                ori_list = line.split()
                write_list = write_line.split()
                for o, w in zip(ori_list, write_list, strict=True):
                    # ignore trailing decimal
                    if o.strip() != w.strip() and abs(float(o.strip()) - float(w.strip())) > 1e-4:
                        raise ValueError(f"{ori_list} != {write_list}")

            assert f"{s.name}\n{atoms}" == snapshot
