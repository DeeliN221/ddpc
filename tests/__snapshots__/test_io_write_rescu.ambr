# serializer version: 1
# name: test_write
  '''
  BN.xyz
  Atoms(symbols='NB', pbc=True)
  '''
# ---
# name: test_write.1
  '''
  Si.xyz
  Atoms(symbols='Si2', pbc=True, initial_magmoms=...)
  '''
# ---
# name: test_write.2
  '''
  SiTi.xyz
  Atoms(symbols='Si254Ti54', pbc=True)
  '''
# ---
# name: test_write.3
  '''
  graphene.xyz
  Atoms(symbols='C2', pbc=True)
  '''
# ---
# name: test_write.4
  '''
  huge.xyz
  Atoms(symbols='C9842B9842N9842', pbc=True)
  '''
# ---
